File: processor/asm/function.py
from collections import namedtuple

Function = namedtuple('Function', [
    'text_glabels',           # List of glabels in the text section
    'asm_conts',              # List of raw assembly contents
    'late_rodata_dummy_bytes', # Bytes of dummy late rodata values
    'jtbl_rodata_size',       # Size of jump table rodata
    'late_rodata_asm_conts',  # List of late rodata contents
    'fn_desc',                # Function descriptor (type, name)
    'data',                   # Raw data (including rodata)

File: src/asm/function.rs
use std::collections::HashMap;

#[derive(Debug, Clone)]
pub struct Function {
    pub text_glabels: Vec<String>,
    pub asm_conts: Vec<String>,
    pub late_rodata_dummy_bytes: Vec<String>,
    pub jtbl_rodata_size: usize,
    pub late_rodata_asm_conts: Vec<String>,
    pub fn_desc: String,
    pub data: HashMap<String, (String, usize)>,
}
